controllers/argocd_metrics_controller.go

```diff
@@ -274,6 +282,42 @@ func (r *ArgoCDMetricsReconciler) createServiceMonitorIfAbsent(namespace string,
 	return err
 }
 
+func (r *ArgoCDMetricsReconciler) reconcileOperatorMetricsServiceMonitor(reqLogger logr.Logger) error {
+
+	data, err := os.ReadFile(operatorPodNamespacePath)
+	if err != nil {
+		reqLogger.Error(err, "Error retrieving operator's running namespace")
+		return err
+	}
+
+	operatorNS := string(data)
+	desiredMetricsServerName := operatorMetricsServiceName + "." + operatorNS + ".svc"
+
+	existingServiceMonitor := &monitoringv1.ServiceMonitor{}
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: operatorMetricsMonitorName, Namespace: operatorNS}, existingServiceMonitor)
+
+	if err == nil {
+		changed := false
+
+		if existingServiceMonitor.Spec.Endpoints[0].TLSConfig.ServerName != desiredMetricsServerName {
```

Comment:
Can `.Spec.Endpoints` be `nil` or point to an empty array ?

Reply:
not unless someone manually deletes the endpoint from the service monitor for some reason
the sevice monitor that we deploy at install time already has an endpoint configured out of the box